packages/scene-composer/src/augmentations/components/three-fiber/anchor/AnchorWidget.tsx

```diff
@@ -113,16 +106,57 @@ export function AsyncLoadedAnchorWidget({
   useEffect(() => {
     setParent(node.parentRef ? getObject3DFromSceneNodeRef(node.parentRef) : undefined);
   }, [node.parentRef]);
-
   // Evaluate visual state based on data binding
   const visualState = useMemo(() => {
-    getCustomIconColor(ruleResult, setOverrideCustomColor);
-    const ruleTargetInfo = getSceneResourceInfo(ruleResult as string);
+    const checkTargetExists = typeof ruleResult === 'object' && 'target' in ruleResult;
+    const checkTargetMetadataExists = typeof ruleResult === 'object' && 'targetMetadata' in ruleResult;
```

Comment:
with the property return type defined for evaluateRule function, we don't need those check